cocotb/share/lib/gpi/GpiCommon.cpp

```diff
@@ -119,6 +119,7 @@ void gpi_embed_end(void)
 
 {
     embed_sim_event(SIM_FAIL, "Simulator shutdown prematurely");
+    embed_sim_end();
```

Comment:
Based on the line above, this only seems to cover failed shutdowns - how are successful shutdowns handled?

Reply:
Every interface calls `gpi_embed_end` as a part of their shutdown callback. Perhaps the above line shouldn't be there.